code/datums/elements/toy_talk.dm

```diff
@@ -0,0 +1,28 @@
+/**
+ * Allows people to talk via the item with .l or .r
+ *
+ * Be sure to override [/atom/movable/proc/GetVoice] if you want the item's "voice" to not default to itself
+ */
+/datum/element/toy_talk
+
+/datum/element/toy_talk/Attach(datum/target)
+	. = ..()
+	if(!isitem(target))
+		return ELEMENT_INCOMPATIBLE
+
+	RegisterSignal(target, COMSIG_ITEM_TALK_INTO, PROC_REF(do_talk))
+
+/datum/element/toy_talk/Detach(datum/source, ...)
+	. = ..()
+	UnregisterSignal(source, COMSIG_ITEM_TALK_INTO)
```

Comment:
it's safe to invoke parent then unregister, right? the pattern is typically `UnregisterSignal(x); return ..()` so i just wanna make sure

Reply:
Yes, element Detach parent call doesn't do anything important